### PR TITLE
Delete Uploaded Files

### PR DESCRIPTION
**Pull request by bhassan6621**
'x' button on the left side of every file uploaded file in side panel, that deletes that file in the array and updates the state and file hints

### CODE REVIEW

File: src/editor.css [hunk 1]
```diff
@@ -135,6 +135,7 @@
 .file-list-item .file-list-item-clipped {
   overflow: hidden;
   text-overflow: ellipsis;
+  padding-left: 30px;
```
Comment (alanorozco) [inline src/editor.css]: Let's set this property once on the rule that applies to both `clipped` and `unclipped`.

File: src/editor.css [hunk 2]
```diff
@@ -173,3 +175,8 @@
 .file-list-empty > div {
   margin: 10px 0;
 }
+
+.delete-file-button {
+  z-index:2;
+  position: absolute;
+}
```
Comment (alanorozco) [inline src/editor.css]: Configure VSCode for newline at end of file :)

File: src/editor.js [hunk 3]
```diff
@@ -198,17 +198,22 @@ const FileList = ({files}) =>
       `;
 
 const dispatchInsertFileRef = redispatchAs(g('insert-file-ref'));
+const dispatchDeleteFile = redispatchAs(g('delete-file'));
 
 /**
  * @param {{name: string}} file
```
Comment (alanorozco) [inline src/editor.js]: nit: add `index` param to `jsdoc`:

```js
/**
 * @param {{name: string}} file
 * @param {number} index
 * @return {lit-html/TemplateResult}
 */
```

File: src/editor.css [hunk 4]
```diff
@@ -173,3 +175,8 @@
 .file-list-empty > div {
   margin: 10px 0;
 }
+
+.delete-file-button {
+  z-index:2;
```
Comment (alanorozco) [inline src/editor.css]: Set `cursor: pointer` :)

File: src/editor.css (hunk 1 above)
Comment (bhassan6621) [reply src/editor.css]: would applying the property like this be alright? 

.file-list-item > div:not(.delete-file-button){
  padding-left: 30px;
}

Comment (alanorozco) [reply src/editor.css]: Yeah! You can also do a compound selector for explicitness:

```css
.file-list-item .file-list-item-clipped, 
.file-list-item .file-list-item-unclipped {
  padding-left: 30px;
}
```
If you go this route, make sure the rule is set after other padding rules for `file-list-item .file-list-item-clipped/unclipped`.

**Pull request merged**